homeassistant/components/climate/__init__.py

```diff
@@ -135,6 +143,35 @@
     vol.Required(ATTR_SWING_MODE): cv.string,
 })
 
+STATE_TO_SERVICE = {
+    STATE_ON: SERVICE_TURN_ON,
+    STATE_OFF: SERVICE_TURN_OFF,
+
+}
+
+SERVICE_TO_SCHEMA = {
```

Comment:
Service schemas are already available in the service registry in the core, and can be accessed like `hass.services.async_services`.

Reply:
Ok. Just thought it felt wrong to involve global stuff in the component when the component knows self.

Reply:
It's the component that has registered its services on the core. We shouldn't duplicate that info.

Reply:
We should still parse this in the component, but we can do it simpler.
```py
service_schemas = {
    service: service_obj.schema
    for service: service_obj
    in hass.services.async_services()[domain]}
```